test/sdk-execution-tests/sdk_execution_tests.py

```diff
@@ -0,0 +1,128 @@
+                yaml_path=os.path.join(test_data_dir,
+                                       f'{test_case["module"]}.yaml'),
+                function_name='my_pipeline' if name ==
+                'pipelines' else test_case['module'],
+                arguments=test_case.get('arguments'))
+            for test_case in test_group['test_cases']
+            if test_case['execute'])
+
+    return parameters
+
+
+def wait(run_result: client.client.RunPipelineResult) -> kfp_server_api.ApiRun:
+    return kfp_client.wait_for_run_completion(
+        run_id=run_result.run_id, timeout=int(TIMEOUT_SECONDS))
+
+
+def import_obj_from_file(python_path: str, obj_name: str) -> Any:
+    sys.path.insert(0, os.path.dirname(python_path))
+    module_name = os.path.splitext(os.path.split(python_path)[1])[0]
+    module = __import__(module_name, fromlist=[obj_name])
+    if not hasattr(module, obj_name):
+        raise ValueError(
+            f'Object "{obj_name}" not found in module {python_path}.')
+    return getattr(module, obj_name)
+
+
+def run(test_case: TestCase) -> Tuple[str, client.client.RunPipelineResult]:
+    full_path = os.path.join(PROJECT_ROOT, test_case.module_path)
+    pipeline_func = import_obj_from_file(full_path, test_case.function_name)
+    run_result = kfp_client.create_run_from_pipeline_func(
+        pipeline_func,
+        enable_caching=True,
+        arguments=test_case.arguments,
+    )
+    run_url = f'{KFP_ENDPOINT}/#/runs/details/{run_result.run_id}'
+    print(
+        f'- Created run {test_case.name}\n\tModule: {test_case.module_path}\n\tURL: {run_url}\n'
+    )
+    return run_url, run_result
+
+
+def get_kfp_package_path() -> str:
+    if os.environ.get('PULL_NUMBER') is not None:
+        path = f'git+https://github.com/kubeflow/pipelines.git@refs/pull/{os.environ.get("PULL_NUMBER")}/merge#subdirectory=sdk/python'
+    else:
+        path = 'git+https://github.com/kubeflow/pipelines.git@master#subdirectory=sdk/python'
+    print(f'Using the following KFP package path for tests: {path}')
+    return path
+
+
+partial_component_decorator = functools.partial(
+    dsl.component, kfp_package_path=get_kfp_package_path())
+
+
+@pytest.mark.asyncio_cooperative
+@pytest.mark.parametrize('test_case', create_test_case_parameters())
+async def test(test_case: TestCase, mocker) -> None:
+    """Asynchronously runs all samples and test that they succeed."""
+    mocker.patch.object(dsl, 'component', partial_component_decorator)
+
+    event_loop = asyncio.get_running_loop()
+    try:
+        run_url, run_result = run(test_case)
+    except Exception as e:
+        raise RuntimeError(
+            f'Error triggering pipeline {test_case.name}.') from e
+
+    api_run = await event_loop.run_in_executor(None, wait, run_result)
+    assert api_run.run.status == 'Succeeded', f'Pipeline {test_case.name} ended with incorrect status: {api_run.run.status}. More info: {run_url}'
```

Comment:
Out of scope for this PR, but want to call out that this is limited comparing to the current sample v2 tests. Not only that we lost checking pipeline specific result, but also that we would not be able to include pipeline that intended to fail in this test suite, e.g. : exit handler use cases.

Reply:
Agreed. We can parameterize for some of this (e.g., `expect_failure: true`), but this doesn't scale well to lots of tests. I am in favor of adding this in the future if we begin seeing pipelines that "`Succeeded`", but with unexpected/incorrect behavior.